packages/tenants/src/context/TenantAgentContextProvider.ts

```diff
@@ -145,4 +168,55 @@ export class TenantAgentContextProvider implements AgentContextProvider {
       await this.registerRecipientKeyForTenant(contextCorrelationId, recipientKey)
     })
   }
+
+  /**
+   * Method to allow updating the tenant storage, this method can be called from the TenantsApi
+   * to update the storage for a tenant manually
+   */
+  public async updateTenantStorage(tenantRecord: TenantRecord, updateOptions?: UpdateAssistantUpdateOptions) {
+    await this.tenantSessionCoordinator.getContextForSession(tenantRecord, {
+      // runInMutex allows us to run the updateTenantStorage method in a mutex lock
+      // prevent other sessions from being started while the update is in progress
+      runInMutex: (agentContext) => this._updateTenantStorage(tenantRecord, agentContext, updateOptions),
+    })
+  }
+
+  /**
+   * Handle the case where the tenant storage is outdated. If auto-update is disabled we will throw an error
+   * and not update the storage. If auto-update is enabled we will update the storage.
+   *
+   * When this method is called we can be sure that we are in the mutex runExclusive lock and thus other sessions
+   * will not be able to open a session for this tenant until we're done.
+   *
+   * NOTE: We don't support multi-instance locking for now. That means you can only have a single instance open and
```

Comment:
Interesting note. I think this applies also for single-tenancy cases where we have multiple instances. It would be another reason of disabling automatic updating in server side deployments.

Reply:
Yes agreed. It is something we can fix in a future version but you'd need something like redis to create the lock across all instances. Or we can use the askar record locking (forUpdate=true) but i'm not really sure yet how that works.

But you generally want all sessions closed when updating so if you have multiple instances in single tenancy it's similar to having one instance during the update. 

Definitely room for improvement here.. Will try to tinker with it more